include/lsst/afw/geom/detail/transformUtils.h

```diff
@@ -0,0 +1,116 @@
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_GEOM_TRANSFORMUTILS_H
+#define LSST_AFW_GEOM_TRANSFORMUTILS_H
+
+#include <memory>
+#include <vector>
+
+#include "astshim.h"
+#include "ndarray.h"
+
+#include "lsst/afw/coord/Coord.h"
+#include "lsst/afw/geom/Endpoint.h"
+#include "lsst/afw/geom/Transform.h"
+#include "lsst/daf/base/PropertyList.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+namespace detail {
+
+/**
+ * Deserialize a Transform from an input stream
+ *
+ * @tparam Transform  the Transform class; can be Transform<FromEndpoint, ToEndpoint>, SkyWcs,
+ * or any other compatible class, i.e. it must support the following (see Transform.h for details):
+ * - a constructor that takes an ast::FrameSet
+ * - static method getShortClassName
+ * - method getFrameSet
+ *
+ * @param[in] is  input stream from which to deserialize this Transform
+ */
+template<class Transform> 
+Transform readStream(std::istream & is);
+
+/**
+ * Serialize a Transform to an output stream
+ *
+ * Version 1 format is as follows:
+ * - The version number (an integer)
+ * - A space
+ * - The short class name, as obtained from getShortClassName
+ * - A space
+ * - The contained ast::FrameSet written using FrameSet.show(os, false)
+ *
+ * @param[out] os  output stream to which to serialize this Transform
+ * @param[in] transform  Transform to serialize
+ */
+template<class Transform>
+void writeStream(Transform const & transform, std::ostream & os);
+
+/*
+ * Provide definitions here in the header file to avoid the need for explicit instantiations
+ */
+
+template<class Transform> 
+Transform readStream(std::istream & is) {
+    int version;
+    is >> version;
+    if (version != 1) {
+        throw LSST_EXCEPT(pex::exceptions::InvalidParameterError,
+                          "Unsupported version " + std::to_string(version));
+    }
+    std::string shortClassName;
+    is >> shortClassName;
+    if (shortClassName != Transform::getShortClassName()) {
+        std::ostringstream os;
+        os << "Class name in stream " << shortClassName << " != " << Transform::getShortClassName();
+        throw LSST_EXCEPT(pex::exceptions::InvalidParameterError, os.str());
+    }
+    auto astStream = ast::Stream(&is, nullptr);
+    auto astObjectPtr = ast::Channel(astStream).read();
+    auto frameSet = std::dynamic_pointer_cast<ast::FrameSet>(astObjectPtr);
+    if (!frameSet) {
+        std::ostringstream os;
+        os << "The AST serialization was read as a " << astObjectPtr->getClassName()
+           << " instead of a FrameSet";
+        throw LSST_EXCEPT(pex::exceptions::InvalidParameterError, os.str());
+    }
+
+    return Transform(*frameSet);
+}
+
+template<class Transform>
+void writeStream(Transform const & transform, std::ostream & os) {
+    os << "1 " << Transform::getShortClassName();
```

Comment:
Please add a `serializationVersion` (or other name, perhaps just `version`) `constexpr` member for the version magic number.